### PR TITLE
Flexible normalization layers

### LINKED ISSUE #31: Specifying normalization layers.
### Is your feature request related to a problem? Please describe.

Currently, the processor is implemented with LayerNormalization. I would like to use other normalization layers (https://pytorch.org/docs/stable/nn.html#normalization-layers) including custom normalization layers. 

### Describe the solution you'd like

I would like to specify the normalization layer of the processor in the config, e.g. transformer.yaml: 
```
layer_norm:  # This needs to be a partial instantiation since it is used in multiple places
  _target_: torch.nn.LayerNorm 
  _partial_: True
  normalized_shape: ${model.num_channels}

processor:
  _target_: anemoi.models.processor.TransformerProcessor
  _convert_: all
  activation: ${model.activation}
  num_layers: 16
  num_chunks: 2
  mlp_hidden_ratio: 4 # GraphTransformer or Transformer only
  num_heads: 16 # GraphTransformer or Transformer only
  window_size: 512
  dropout_p: 0.0 # GraphTransformer
  layer_norm: ${model.layer_norm} # (Optional) Default nn.LayerNorm
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Organisation

_No response_

### PR DESCRIPTION
**Pull request by jakob-schloer**
This PR combines the issue https://github.com/ecmwf/anemoi-core/issues/31 with the PR https://github.com/ecmwf/anemoi-models/pull/35 by @cathalobrien.

# Describe your changes

This PR makes it possible to switch the implementation of Linear and LayerNorm kernels in the config. 

At the moment we use torch.NN implementation for many layers in Anemoi model e.g. torch.nn.layerNorm, torch.NN.linear. This has the advantage of being available out of the box with torch and portable to many systems (CPU, AMD and Nvidia GPUs). However, other layer implementations might be more efficient for certain hardware, or we might want to use a custom layer. 

This PR adds the following block to config/model/<model>.yaml:
```yaml
  layer_kernels:
    LayerNorm:
      #_target_: "transformer_engine.pytorch.LayerNorm"
      _target_: "liger_kernel.transformers.rms_norm.LigerRMSNorm"
      #_target_: "torch.nn.LayerNorm" #the default PyTorch implementation
      _partial_: True
      #Any arguments to your chosen function go here e.g.
      #bias: False
    Linear:
      #_target_: "transformer_engine.pytorch.Linear"
      _target_: "torch.nn.Linear"
      _partial_: True
```
You can pass any parameters to your new kernels in the config file, after "_partial_ : True". Hydra tries to load the desired kernel in "models/encoder_processor_decoder.py". If the desired library isn't available, torch currently will fall back to torch.nn.<kernel>.

The calls to `torch.nn` are then replaced with
```python
- self.layer_norm = nn.LayerNorm(normalized_shape=num_channels)
+ self.layer_norm = layer_kernels['LayerNorm'](normalized_shape=num_channels)
```

In the future, this syntax could be extended to replace other layers if required. More specifically, a follow up PR that includes conditional layer norm.

# Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# Checklist before requesting a review

- [X] I have performed a self-review of my code
- [X] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation and docstrings to reflect the changes
- [ ]  have added tests that prove my fix is effective or that my feature works
- [ ] I have ensured that the code is still pip-installable after the changes and runs
- [x] I have not introduced new dependencies in the inference partion of the model
- [X] I have ran this on single GPU
- [X] I have ran this on multi-GPU or multi-node
- [ ] I have ran the Benchmark Profiler against the old version of the code
- [x] I've tested the changes with the Transformer, GraphTransformer and GNN

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-models?pullRequest=95) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-models?pullRequest=95) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: cathalobrien<br/>:x: jakob-schloer<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/anemoi-models?pullRequest=95) it.</sub>
